sdk/communication/azure-communication-administration/src/main/java/com/azure/communication/administration/PhoneNumberAsyncClient.java

```diff
@@ -819,19 +819,82 @@
                 });
     }
 
-    private BiFunction<PollingContext<PhoneNumberSearch>,
-        PollResponse<PhoneNumberSearch>, Mono<PhoneNumberSearch>> 
+    private BiFunction<PollingContext<PhoneNumberReservation>,
+        PollResponse<PhoneNumberReservation>, Mono<PhoneNumberReservation>>
         cancelSearchOperation() {
         return (pollingContext, firstResponse) -> {
             cancelSearch(pollingContext.getLatestResponse().getValue().getSearchId());
             return Mono.just(pollingContext.getLatestResponse().getValue());
         };
     }
 
-    private Function<PollingContext<PhoneNumberSearch>,
-        Mono<PhoneNumberSearch>> createSearchFetchResultOperation() {
+    private Function<PollingContext<PhoneNumberReservation>,
+        Mono<PhoneNumberReservation>> createSearchFetchResultOperation() {
         return pollingContext -> {
             return Mono.just(pollingContext.getLatestResponse().getValue());
         };
     }
+
+    /**
+     * LongRunningOperation for the purchaseSearch.
+     *
+     * @param reservationId ID of the reservation
+     * @param pollInterval Time lapse of the poll request.
+     * @return A {@link PollerFlux} object.
+     */
+
+    @ServiceMethod(returns = ReturnType.COLLECTION)
+    public PollerFlux<Void, Void> beginPurchaseReservation(String reservationId, Duration pollInterval) {
+        Objects.requireNonNull(reservationId, "'reservationId' can not be null.");
+        Objects.requireNonNull(pollInterval, "'pollInterval' can not be null.");
+
+        return new PollerFlux<Void, Void>(pollInterval,
+            purchaseReservationActivationOperation(reservationId),
+            purchaseReservationCreatePollOperation(reservationId),
+            (activationResponse, pollingContext) -> Mono.error(new RuntimeException("Cancellation is not supported")),
+            purchaseReservationFetchResultOperation());
+    }
+
+    private Function<PollingContext<Void>, Mono<Void>> purchaseReservationActivationOperation(String reservationId) {
+
+        return (pollingContext) -> {
+            Mono<Void> response = purchaseSearchWithResponse(reservationId).map(
+                purchaseResponse -> purchaseResponse.getValue());
+            return response;
+        };
+    }
+
+    private Function<PollingContext<Void>, Mono<PollResponse<Void>>>
+                purchaseReservationCreatePollOperation(String reservationId) {
```

Comment:
nit: shouldn't this be purchaseReservationPollOperation (don't understand the 'create' in between)